tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -1942,13 +1943,17 @@ bool DeclResultIdMapper::finalizeStageIOLocations(bool forInput) {
       }
 
       // Make sure the same location is not assigned more than once
-      if (locSet.isLocUsed(loc, idx)) {
-        emitError("stage %select{output|input}0 location #%1 already assigned",
-                  attrLoc)
-            << forInput << loc;
-        noError = false;
+      for (uint32_t l = loc; l < loc + locCount; ++l) {
+        if (locSet.isLocUsed(l, idx)) {
+          emitError(
+              "stage %select{output|input}0 location #%1 already assigned",
```

Comment:
slight nit?:
Would it be more useful to say "location X overlaps with another variable location?"
If I don't know much about locations, and I see "already assigned", I would look at the actual value X being used in my code. Overlaps, I would think there is some kind of range, and search about location ranges?
Not sure.

Reply:
It also could be handy to have a note that refers to the previous declaration.

Reply:
@Keenuts "consumed" is the language used in the spec so I've switched to that -- I think that it reads well now for both cases where it's 1-1 (the norm) and a range (this edge case).
@llvm-beanz Done, thanks!